doc/user-guide.adoc

```diff
@@ -215,3 +215,101 @@ spec:
   disableHTTPRoute: true
 ----
 
+[[scaledown-transaction-recovery]]
+## Transaction recovery during scaledown
+
+As the application deployed in the WildFly application server
+may use JTA transactions there and question emerges: what happen when the cluster size is scaled down.
+On decreasing of number of active WildFly replicas there may be still some in-doubt transactions running.
+When the pod is removed then all the in-progress transactions are stopped and rolled-back.
+A more troublesome situation occurs when XA transactions are used.
+When the XA transaction declares it's prepared it's a promise to finish the transaction successfully.
+But the transaction manager which made this promise is a running process inside of the WildFly server.
+Then just simply shutting down such pod would lead to a data inconsistencies or to fact of leaving locks on data.
+
+That's why it's needed to be ensured that all transactions are finished before the number of replicas is really decreased.
+For that purpose the WildFly operator provides scale down functionality which verifies if all transactions were finished
+and only then marks the pod as clean for termination.
+
+Decreasing the replica size in the `WildFlyServer` customer resource is done at field `WildFlyServer.Spec.Replicas` (see <<size>>).
+You can use for example patch command like
+
+```
+oc patch wildflyserver <<name>> -p '[{"op":"replace", "path":"/spec/replicas", "value":0}]' --type json
+```
+
+or you can manually edit and change the replica number with with `oc edit wildflyserver <<name>>`.
+
+NOTE: Decreasing replica size at the `StatefulSet` or deleting the Pod itself has no effect and as such changes will be reverted.
+
+WARNING: if you decide to delete whole `WildflyServer` definition (`oc delete wildflyserver <<deployment_name>>`)
+         then no transaction recovery process is started and the pod is terminated regardless of unfinished transactions.
+         If you want to remove the deployment and you expects doing it in transaction safe way
```

Comment:
"If you want to remove the deployment in a safe way without data inconsistencies, you need..."